batavia/builtins/max.js

```diff
@@ -39,7 +39,7 @@ function max(args, kwargs) {
     try {
         while (true) {
             var next = iterobj.__next__()
-            if (next.__gt__(max)) {
+            if (next.__gt__(max).valueOf()) {
```

Comment:
I dont know, why this caused problem **only** with nested tuples... But it helps.